addon/templates/components/common/bs-dropdown/button.hbs

```diff
@@ -0,0 +1,21 @@
+<button
+  disabled={{this.__disabled}}
+  type={{this.buttonType}}
+  title={{this.title}}
+  aria-expanded={{this.aria-expanded}}
+  class="btn dropdown-toggle {{if this.active "active"}} {{if this.block "btn-block"}} {{this.sizeClass}} {{this.typeClass}}"
+  ...attributes
+  {{on "click" this.handleClick}}
+  {{on "keydown" this.handleKeyDown}}
+  {{ref @dropdown "toggleElement"}}
```

Comment:
Yielding the full component and changing it's content looks scary to me. Did you consider yielding a `registerToggleElement` method that is used here with `{{did-insert}}` modifier?

```js
  @action
  registerToggleElement(element) {
    assert(element instanceof HTMLElement);

    this.set('toggleElement', element);
  }
```

```hbs
  {{yield
    registerToggleElement=this.registerToggleElement
  }}
```

```hbs
  {{did-insert @registerToggleElement}}
```

Reply:
:+1:

We basically did set it directly also before that change, so the new code did not change that. But this certainly looks better! We do use this pattern already in the parent/child mixins, but should probably stop using these mixins for new code...

Reply:
Done, including unregistering.